app/controllers/api/v1/portfolio_items_controller.rb

```diff
@@ -24,6 +24,7 @@ def index
       end
 
       def create
+        params.require(:portfolio_id)
```

Comment:
@eclarizio We don't need this anymore since the common gem does the validation on the required input parameters.

Reply:
@mkanoor Can you show me where it does this? All I could find was [params_for_create](https://github.com/RedHatInsights/insights-api-common-rails/blob/27f0e5c1867580c043e7dd3a1be99390d78d1f93/lib/insights/api/common/application_controller_mixins/parameters.rb#L10) which then calls [body_params](https://github.com/RedHatInsights/insights-api-common-rails/blob/d57bf92e39eec1b16bd9538f5e4411fcacadb540/lib/insights/api/common/application_controller_mixins/request_body_validation.rb#L19) and just `.permit!` which permits everything, no?

Reply:
We validate every request coming for POST/PATCH and validate it against the openapi definitions here
https://github.com/RedHatInsights/insights-api-common-rails/blob/d57bf92e39eec1b16bd9538f5e4411fcacadb540/lib/insights/api/common/application_controller_mixins/request_body_validation.rb#L40

And the validate_request is called from here
https://github.com/RedHatInsights/insights-api-common-rails/blob/d57bf92e39eec1b16bd9538f5e4411fcacadb540/lib/insights/api/common/application_controller_mixins/request_body_validation.rb#L14

The work is done by the openapi parser gem
https://github.com/ota42y/openapi_parser

We frequently contribute to this gem and the logic is shared with all of our micro services. With the openapi spec being the single source of truth we don't need much of the checks in our controllers.

Reply:
@mkanoor Cool thanks, that helped me figure out that the way I had changed the openapi json file was incorrect because my test was failing when I took out the `params.require`.

It's fixed now 👍